pkg/upgrade/ready_test.go

```diff
@@ -0,0 +1,286 @@
+package upgrade
+
+import (
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestIsPodReady(t *testing.T) {
+	tests := []struct {
+		name string
+		pod  *corev1.Pod
+		want bool
+	}{
+		{
+			name: "ready",
+			want: true,
+			pod: &corev1.Pod{
+				Status: corev1.PodStatus{
+					Conditions: []corev1.PodCondition{
+						{
+							Type:   corev1.PodInitialized,
+							Status: corev1.ConditionTrue,
+						},
+						{
+							Type:   corev1.PodReady,
+							Status: corev1.ConditionTrue,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "missing",
+			want: false,
+			pod: &corev1.Pod{
+				Status: corev1.PodStatus{
+					Conditions: []corev1.PodCondition{
+						{
+							Type:   corev1.PodInitialized,
+							Status: corev1.ConditionFalse,
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		if got := isPodReady(tt.pod); got != tt.want {
+			t.Errorf("isPodReady(%s) = %v, want %v", tt.name, got, tt.want)
+		}
+	}
+}
+
+func TestNodeReady(t *testing.T) {
```

Comment:
yikes, we have a nodeIsReady function and a isNodeReady function :(
can you refactor isNodeReady into nodeIsReady and condense the testing?